zebra-chain/src/sapling/arbitrary.rs

```diff
@@ -99,3 +102,43 @@ impl Arbitrary for OutputInTransactionV4 {
 
     type Strategy = BoxedStrategy<Self>;
 }
+
+/// A wrapper for a RedJubJub spending verification key.
+///
+/// This is used by proptests since we can't implement Arbitrary for the
+/// redjubjub::VerificationKeyBytes (it's in another crate).
+/// We then implement it for the wrapper type instead.
+#[derive(Debug)]
+struct SpendVerificationKeyBytesWrapper(redjubjub::VerificationKeyBytes<redjubjub::SpendAuth>);
+
+impl Arbitrary for SpendVerificationKeyBytesWrapper {
+    type Parameters = ();
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (prop::array::uniform32(any::<u8>()))
+            .prop_map(|bytes| {
+                let mut rng = ChaChaRng::from_seed(bytes);
+                let sk = redjubjub::SigningKey::<redjubjub::SpendAuth>::new(&mut rng);
+                let pk = redjubjub::VerificationKey::<redjubjub::SpendAuth>::from(&sk);
+                SpendVerificationKeyBytesWrapper(pk.into())
+            })
+            .boxed()
+    }
+
+    type Strategy = BoxedStrategy<Self>;
+}
+
+impl Arbitrary for tree::Root {
+    type Parameters = ();
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (vec(any::<u8>(), 64))
+            .prop_map(|bytes| {
+                let bytes = bytes.try_into().expect("vec is the correct length");
+                bls12_381::Scalar::from_bytes_wide(&bytes).to_repr().into()
+            })
+            .boxed()
+    }
+
+    type Strategy = BoxedStrategy<Self>;
+}
```

Comment:
The bytes that `sapling::tree::Root` currently wraps are actually an instance of `jubjub::Fq`, which is an [alias for `bls12_381::Scalar`](https://github.com/zkcrypto/jubjub/blob/main/src/lib.rs#L56), so we don't need to import `bls12_381` for this, we can just do another `from_bytes_wide` here:

```suggestion
impl Arbitrary for tree::Root {
    type Parameters = ();

    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
        (vec(any::<u8>(), 64))
            .prop_map(|bytes| {
                jubjub::Fq::from_bytes_wide(&bytes).into()
            })
            .boxed()
    }

    type Strategy = BoxedStrategy<Self>;
}
```

Reply:
Nice, thanks! Added in 29b13a0, had to tweak it a bit